File: models/java/dog_model.java
class Dog {
    public String name;
    public int size;
    public int age;

    public Dog(String name, int size, int age) {
        this.name = name;
        this.size = size;
        this.age = age;
    }

    public String bark() {
       if (size >= 50) return "Wooof! Woof!";
       else if (size >= 20) return "Ruff! Ruff!";
       else return "Yip! Yip!";
    }

    public int calcHumanAge() {
        return (age - 1) * 7 + 12;
    }
}

class MyClass{
    public static void main(String[] args){
        Dog goldenRetriever = new Dog("Golden Retrieber", 60, 10);
        System.out.println(goldenRetriever.bark());
        System.out.println(goldenRetriever.calcHumanAge());

        Dog siberianHusky = new Dog("Siberian Husky", 55, 6);
        System.out.println(siberianHusky.bark());
        System.out.println(siberianHusky.calcHumanAge());

        Dog poodle = new Dog("poodle", 10, 1);
        System.out.println(poodle.bark());
        System.out.println(poodle.calcHumanAge());

        Dog shibainu = new Dog("shibainu", 35, 4);
        System.out.println(shibainu.bark());
        System.out.println(shibainu.calcHumanAge());

    }
}

File: models/python/dog_model.py
'''
String name	犬の種類の名前
int size	犬のサイズ
int age	犬の年齢
String bark()	犬の鳴き声を文字列として返します。犬のサイズが 50 以上の時、Wooof! Woof!、サイズが 20 以上の時、Ruff! Ruff!、それ以外の時は、Yip! Yip! と返します。
int calcHumanAge()	犬の年齢から人間の年齢に換算します。人間の年齢＝ 12 +（犬の年齢 - 1）× 7 を使用してください。
'''
class Dog():
    def __init__(self, name, size, age):
        self.name = name
        self.size = size
        self.age = age

    def bark(self):
        if self.size >= 50:
            return 'Wooof! Woof!'
        elif self.size >= 20:
            return 'Ruff! Ruff!'
        else:
            return 'Yip! Yip!'

    def calcHumanAge(self):
        return 12 + (self.age - 1) * 7

goldenRetriever = Dog("Golden Retriever", 60, 10)
print(goldenRetriever.bark())
print(goldenRetriever.calcHumanAge())

siberianHusky = Dog("Siberian Husky", 55, 6)
print(siberianHusky.bark())
print(siberianHusky.calcHumanAge())

poodle = Dog("poodle", 10, 1)
print(poodle.bark())
print(poodle.calcHumanAge())

shibaInu = Dog("shibaInu", 35, 4)
print(shibaInu.bark())
print(shibaInu.calcHumanAge())
